components/sst_importer/src/import_file.rs

```diff
@@ -413,17 +438,72 @@ impl ImportDir {
         Ok(())
     }
 
-    pub fn list_ssts(&self) -> Result<Vec<SstMeta>> {
+    fn fill_by_persisted_meta(
+        &self,
+        p: &Path,
+        sc: Option<&DataKeyManager>,
+        m0: &mut SstMeta,
+    ) -> Result<()> {
+        use std::io::{Error as IoErr, ErrorKind as IoErrs};
+        let fname = p
+            .file_name()
+            .ok_or_else(|| Error::InvalidSstPath(p.to_owned()))?;
+        let meta_path = self
+            .meta_dir
+            .join(self.get_meta_file_name(&fname.to_string_lossy()));
+
+        let meta_bytes = match sc {
+            None => file_system::read(&meta_path)?,
+            Some(s) => {
+                let mut v = Vec::new();
+                s.open_file_for_read(&meta_path)?.read_to_end(&mut v)?;
+                v
+            }
+        };
+        let old_uuid = m0.get_uuid().to_owned();
+        let backup = m0.clone();
+        m0.merge_from_bytes(&meta_bytes)
+            .map_err(|err| IoErr::new(IoErrs::InvalidData, err))?;
+        if old_uuid != m0.get_uuid() {
+            // The meta might be modified by the `merge_from_bytes`, let's
+            // restore it.
+            *m0 = backup;
+            return Err(Error::FileCorrupted(
+                meta_path,
+                format!(
+                    "the meta file uuid doesn't match the filename: {:?} vs {:?}",
+                    old_uuid,
+                    m0.get_uuid()
+                ),
+            ));
+        }
+        Ok(())
+    }
+
+    fn get_meta_file_name(&self, n: &str) -> String {
+        format!("{n}.meta.wire")
+    }
+
+    fn get_meta_by_path(&self, p: &Path, km: Option<&DataKeyManager>) -> Result<SstMeta> {
+        let mut m0 = parse_meta_from_path(p)?;
+        if let Err(err) = self.fill_by_persisted_meta(p, km, &mut m0) {
```

Comment:
Does it cost a lot of time? Afraid of blocking raftstore for a long time. Can we read it when necessary in cleanup sst worker?